apps/condo/domains/user/schema/ResetUserService.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * Generated by `createservice user.ResetUserService --type mutations`
+ */
+
+/**
+ * Reset user mutation removes phone, email and password from user, making it impossible to use this user.
+ *
+ * Mutation also sets name to "Deleted User" and do not set deletedAt.
+ * Reason for this is the fact that if we soft-delete user, then we are about to face a lot of "user is null" kind of errors in unpredictable places.
+ * Name is changed so we can distinguish reset users.
+ *
+ * Telegram uses the same approach: if user is deleted, then all his messages and other content are available, but his name is now "Deleted user"
+ *
+ * Primary use cases:
+ * 1. QA Wants to test SMS services, but dont want to get multiple phone numbers
+ * 2. User wants to remove all his data from our system.
+ */
+
+const { GQLCustomSchema, getById } = require('@core/keystone/schema')
+
+const access = require('@condo/domains/user/access/ResetUserService')
+const { DELETED_USER_NAME } = require('@condo/domains/user/constants')
+const { User } = require('@condo/domains/user/utils/serverSchema')
+
+const { DV_UNKNOWN_VERSION_ERROR, NOT_FOUND_ERROR } = require('@condo/domains/common/constants/errors')
+
+
+const ResetUserService = new GQLCustomSchema('ResetUserService', {
+    types: [
+        {
+            access: true,
+            type: 'input ResetUserInput { dv: Int!, sender: JSON! id: String!}',
+        },
+        {
+            access: true,
+            type: 'type ResetUserOutput { status: String! }',
+        },
+    ],
+    
+    mutations: [
+        {
+            access: access.canResetUser,
+            schema: 'resetUser(data: ResetUserInput!): ResetUserOutput!',
+            resolver: async (parent, args, context, info, extra = {}) => {
+                const { data } = args
+                const { dv, id } = data
+
+                if (dv !== 1) {
+                    throw new Error(`${DV_UNKNOWN_VERSION_ERROR}dv] Unknown \`dv\``)
+                }
+
+                const user = await getById('User', id)
+                if (!user) {
+                    throw new Error(`${NOT_FOUND_ERROR}user] No user found for this id`)
+                }
+
+                const adminContext = await context.createContext({ skipAccessControl: true })
+                await User.update(adminContext, id, {
+                    phone: null,
```

Comment:
Setting all user identical to null = no user in db...
Since if we clean UserHistory we cannot restore it...
So now it's equal to deletedAt performance...

May be @pahaz can resolve our conversation?

Reply:
No, it's not identical, since deletedAt means that we will delete user from db at some point. In this case we do not delete user from db

This solution was initially proposed by @pahaz, as it is also used in telegram

Reply:
Really we have two cases:
 1) we need to release the existing user phones for test purposes
 2) we need to arise user data by user request

It's hard to really remove use because of lots of FKs. For that reason, we use things like that. Erase the user data and not change the business logic